CognitoUserToCSV.py

```
@@ -14,79 +16,151 @@
+    if file_path.parent != Path('.'):
+        try:
+            file_path.parent.mkdir(parents=True, exist_ok=True)
+            print(Fore.CYAN + f"INFO: Created directory: {file_path.parent}")
+        except Exception as e:
+            print(Fore.RED + f"ERROR: Cannot create directory {file_path.parent}: {str(e)}")
+            exit(1)
+
+    # Validate filename
+    try:
+        # Test if we can create/write to the file
+        test_file = open(file_path, 'w')
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)